### PR TITLE
NIP-01 Reword filter limits

### LINKED ISSUE #1115: NIP-01: `limit` in multiple filters
How should relays interpret the `limit` specification if present in multiple filters? There seem to be contradicting specs regarding the `limit` filter field if given by multiple filters

```json
[
  {"limit": 2, …},
  {"limit": 40, …}
]
```

1. NIP-01 specifies that we can use multiple filters in a subscription

  > A REQ message may contain multiple filters. In this case, events that match any of the filters are to be returned, i.e., multiple 
  filters are to be interpreted as || conditions.

2. NIP-01 also specifies that a `limit` determines the `<maximum number of events relays SHOULD return in the initial query>`

Is the limit to be applied for each filter? So we return 40-42 events depending on whether the matching events overlap.
Or should the maximum `limit` of any given limit be relevant, since the idea is to clamp the event count of the initial query?

### PR DESCRIPTION
**Pull request by schmijos**
Fixes #1115 by uniform glossary and using "each"

### CODE REVIEW

File: 01.md [hunk 1]
```diff
@@ -127,13 +127,13 @@ Clients can send 3 types of messages, which must be JSON arrays, according to th
...
 
-Filter attributes containing lists (`ids`, `authors`, `kinds` and tag filters like `#e`) are JSON arrays with one or more values. At least one of the arrays' values must match the relevant field in an event for the condition to be considered a match. For scalar event attributes such as `authors` and `kind`, the attribute from the event must be contained in the filter list. In the case of tag attributes such as `#e`, for which an event may have multiple values, the event and filter condition values must have at least one item in common.
+Filter attributes containing lists (`ids`, `authors`, `kinds` and tag conditions like `#e`) are JSON arrays with one or more values. At least one of the arrays' values must match the relevant field in an event for the condition to be considered a match. Scalar event attributes such as `pubkey` and `kind` must match the corresponding filter condition exactly. In the case of tag attributes such as `#e`, for which an event may have multiple values, the event and filter condition values must have at least one item in common.
```
Comment (schmijos) [inline 01.md:136]: `authors` was wrong here. Replaced with `pubkey`.

Comment (fiatjaf) [reply 01.md:136]: This sounds more confusing than before.

If there is a filter `{"kinds": [1,2,3]}` how does an event `{"kind": 1}` match it exactly?

Comment (schmijos) [reply 01.md]: I see your point. I deleted the sentence and included the information in the filter format example.

**Comment by schmijos** [comment]
Closing for lack of interest.

**Comment by bezysoftware** [comment]
Oh no, this actually makes a lot of sense, I was also struggling with the filters wording especially around the `limit` field.

**Pull request closed without merging**